packages/recoil/recoil_values/Recoil_atom.js

```diff
@@ -292,6 +293,23 @@ function baseAtom<T>(options: BaseAtomOptions<T>): RecoilState<T> {
         return getLoadable(recoilValue).toPromise();
       }
 
+      function observe<S>(
+        recoilValue: RecoilValue<S>,
+        observer: (value: S) => void,
+      ): () => void {
```

Comment:
Should probably detect and error that this (and `onSet()` as well) can only be called directly from the effect (e.g. `duringInit`).  Otherwise it could encourage trying to setup dynamic subscriptions and things get more complicated...

Reply:
I added an error condition in onSetValue- I'd be glad to add one in onSet if you prefer, but I don't want to blow the scope of the PR up too much